### PR TITLE
libwebsockets: add version 4.2.0

### PR DESCRIPTION
**Pull request by prince-chrismc**
Generated and committed by [Conan Center Bot](https://github.com/qchateau/conan-center-bot)
Find more updatable recipes in the [GitHub Pages](https://qchateau.github.io/conan-center-bot/)

Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Only GCC-4.9 shared failed =/

**Comment by prince-chrismc** [comment]
They all fail... did something change in the docker images?

**Review by SSE4 — APPROVED** [review]

**Review by ericLemanissier — DISMISSED** [review]

File: recipes/libwebsockets/all/conanfile.py [hunk 1]
```diff
@@ -197,6 +197,12 @@ def configure(self):
             del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
+        
+        if self.options.shared and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+                # https://github.com/conan-io/conan-center-index/pull/5321#issuecomment-826367276
+                raise ConanInvalidConfiguration("{}/{} shared=True with gcc<5 does not build. Please submit a PR with a fix.".format(self.name, self.version))
+        if tools.Version(self.version) == "4.0.15" and self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) >= "12":
+                raise ConanInvalidConfiguration("{}/{} with clang>=12 does not build. Please submit a PR with a fix.".format(self.name, self.version))
```
Comment (mathbunnyru) [inline recipes/libwebsockets/all/conanfile.py:205]:
```suggestion
                raise ConanInvalidConfiguration("{}/{} with apple-clang>=12 does not build. Please submit a PR with a fix.".format(self.name, self.version))
```

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

**Review by mathbunnyru — DISMISSED** [review]

File: recipes/libwebsockets/all/conanfile.py [hunk 2]
```diff
@@ -197,6 +197,12 @@ def configure(self):
             del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
+        
+        if self.options.shared and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+                # https://github.com/conan-io/conan-center-index/pull/5321#issuecomment-826367276
+                raise ConanInvalidConfiguration("{}/{} shared=True with gcc<5 does not build. Please submit a PR with a fix.".format(self.name, self.version))
+        if tools.Version(self.version) == "4.0.15" and self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) >= "12":
```
Comment (uilianries) [inline recipes/libwebsockets/all/conanfile.py:204]:
```suggestion
        if tools.Version(self.version) <= "4.0.15" and self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) >= "12":
```

File: recipes/libwebsockets/all/conanfile.py [hunk 3]
```diff
@@ -197,6 +197,12 @@ def configure(self):
             del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
+        
+        if self.options.shared and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
```
Comment (uilianries) [inline recipes/libwebsockets/all/conanfile.py:201]:
```suggestion
    def validate(self):
        if self.options.shared and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
```

**Review by mathbunnyru — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**